src/clj/rems/db/applications.clj

```diff
@@ -54,69 +52,39 @@
 ;;; Running commands
 
 (defn get-catalogue-item-licenses [catalogue-item-id]
-  (let [item (catalogue/get-localized-catalogue-item catalogue-item-id {})
-        workflow-licenses (-> (workflow/get-workflow (:wfid item))
+  (let [item (rems.db.catalogue/get-catalogue-item catalogue-item-id)
+        resource-licenses (:licenses (rems.db.resource/get-resource (:resource-id item)))
+        workflow-licenses (-> (rems.db.workflow/get-workflow (:wfid item))
                               (get-in [:workflow :licenses]))]
-    (->> (licenses/get-licenses {:items [catalogue-item-id]})
-         (keep-keys {:id :license/id})
-         (into workflow-licenses)
-         (distinct-by :license/id))))
+    (->> (concat resource-licenses workflow-licenses)
+         (map #(clojure.set/rename-keys % {:id :license/id}))
+         (distinct-by :license/id)
+         (into []))))
```

Comment:
here we could:
- rename license ids in relevant line (resource licenses?)
- skip converting to vector (`into []`)  if not needed